src/java.base/share/classes/java/lang/runtime/SwitchBootstraps.java

```diff
@@ -717,6 +721,11 @@ private static Consumer<CodeBuilder> generateTypeSwitchSkeleton(Class<?> selecto
         };
     }
 
+    private static boolean isNotValidPair(Class<?> selectorType, Object caseLabel) {
+        return (selectorType == boolean.class && caseLabel != boolean.class && caseLabel != Boolean.class) ||
```

Comment:
What happens if `caseLabel` is a reference class?

Reply:
`(selectorType == boolean.class && caseLabel != boolean.class && caseLabel != Boolean.class)` -> true so it is not a valid pair. Do you think it is wrong?

Reply:
e.g., https://github.com/openjdk/jdk/pull/25090/commits/7b030bee3e23206c030970f6c24be3bcb5241361